contracts/external/traders/StopLimitOrders.sol

```diff
@@ -502,6 +536,37 @@ contract LimitOrders is
         );
     }
 
+    /**
+     * Verifies that the order is decreasing the size of the maker's position.
+     */
+    function verifyDecreaseOnly(
+        Types.Par memory oldInputPar,
+        Types.Par memory newInputPar,
+        Types.AssetAmount memory assetAmount,
+        Account.Info memory makerAccount,
+        uint256 outputMarketId
+    )
+        private
+        view
+    {
+        // verify inputMarketId
+        Require.that(
+            newInputPar.isZero()
+            || (newInputPar.value <= oldInputPar.value && newInputPar.sign == oldInputPar.sign),
+            FILE,
+            "inputMarket not decreased"
+        );
+
+        // verify outputMarketId
+        Types.Wei memory oldOutputWei = SOLO_MARGIN.getAccountWei(makerAccount, outputMarketId);
```

Comment:
so this will give you the balance before the trade, right?

Reply:
yes